extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -74,6 +75,39 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 			}
 		} catch (e) {
 			vscode.window.showErrorMessage(loc.openFileError(bookPath, e instanceof Error ? e.message : e));
+		} finally {
+			// add file watcher on toc file.
+			fsw.watchFile(path.join(bookPath, '_data', 'toc.yml'), async (curr, prev) => {
+				if (curr.mtime > prev.mtime) {
+					let index = this.books.findIndex(book => book.bookPath === bookPath);
+					await this.books[index].initializeContents().then(() => {
+						this._onDidChangeTreeData.fire(this.books[index].bookItems[0]);
+					});
+					this._onDidChangeTreeData.fire();
+				}
+			});
+		}
+	}
+
+	async closeBook(book?: BookTreeItem): Promise<void> {
```

Comment:
Since we don't do anything if no book is passed in, we should probably just make this required.